Frameworks/Foundation/NSOperationQueue.mm

```diff
@@ -220,65 +369,19 @@ - (NSOperationQueue*)_doMainWork {
 /**
  @Status Interoperable
 */
-- (BOOL)hasMoreWork {
-    if (priv->curOperation != nil) {
-        return TRUE;
-    }
-
-    for (int i = 0; i < NSOperationQueuePriority_Count; i++) {
-        if (priv->queues[i] != NULL) {
-            return TRUE;
-        }
-    }
-
-    return FALSE;
-}
-
-/**
- @Status Interoperable
-*/
 - (id)init {
     if (self = [super init]) {
-        priv = new NSOperationQueuePriv();
-
-        priv->_maxConcurrentOperationCount = 1;
-        priv->workAvailable = [[NSCondition alloc] init];
-        priv->suspendedCondition = [[NSCondition alloc] init];
-        priv->allWorkDone = [[NSCondition alloc] init];
-        priv->isSuspended = 0;
-
         _completionQueue = dispatch_queue_create("NSOperation finish queue", nullptr);
+        _priv = std::make_unique<NSOperationQueuePriv>(self);
```

Comment:
a setName call could be made but I don't really like that. I think unique_ptr is fine here.

---
In reply to: [95451424](https://github.com/Microsoft/WinObjC/pull/1676#discussion_r95451424) [](ancestors = 95451424)